cylc/flow/scripts/graph.py

```diff
@@ -215,19 +243,123 @@ def get_option_parser():
     return parser
 
 
+def dot(opts, lines):
+    """Render a graph using graphviz 'dot'.
+
+    This crudely re-parses the output of the reference output for simplicity.
+
+    This functionality will be replaced by the GUI.
+
+    """
+    if not which('dot'):
+        sys.exit('Graphviz must be installed to render graphs.')
+
+    # set filename and output format
+    if opts.output:
+        filename = opts.output
+        try:
+            fmt = filename.split('.')[1]
+        except IndexError:
+            sys.exit('Output filename requires a format.')
+    else:
+        filename = NamedTemporaryFile().name
+        fmt = 'png'
+
+    # scrape nodes and edges from the reference output
+    node = re.compile(r'node "(.*)" "(.*)"')
+    edge = re.compile(r'edge "(.*)" "(.*)"')
+    nodes = {}
+    edges = []
+    for line in lines:
+        match = node.match(line)
+        if match:
+            if opts.namespaces:
+                task = match.group(1)
+                cycle = ''
+            else:
+                task, cycle = match.group(1).split('.')
+                nodes.setdefault(cycle, []).append(task)
+            continue
+        match = edge.match(line)
+        if match:
+            edges.append(match.groups())
+
+    # write graph header
+    dot = [
+        'digraph {',
+        '  graph [fontname="sans" fontsize="25"]',
+        '  node [fontname="sans"]',
+    ]
+    if opts.transpose:
+        dot.append('  rankdir="LR"')
+    if opts.namespaces:
+        dot.append('  node [shape="rect"]')
+
+    # write nodes
+    for cycle, tasks in nodes.items():
+        if opts.cycles:
+            dot.extend(
+                [
+                    f'  subgraph "cluster_{cycle}" {{ ',
+                    f'    label="{cycle}"',
+                    '    style="dashed"',
+                ]
+            )
+        dot.extend(
+            [
+                rf'    "{task}.{cycle}" [label="{task}\n{cycle}"]'
+                for task in tasks
+            ]
+        )
+        if opts.cycles:
+            dot.append('  }' if opts.cycles else '')
```

Comment:
Should this be
```suggestion
        dot.append('  }' if opts.cycles else '')
```
or
```suggestion
        if opts.cycles:
            dot.append('  }')
```